parsing.go

```diff
@@ -131,50 +153,73 @@ func parseMarkdown(f string) ([]*Req, error) {
 					if level == lastHeadingLevel {
 						return nil, fmt.Errorf("requirement heading on line %d at same level as previous heading on line %d (%d): %q", lno, lastHeadingLine, level, line)
 					}
-					start = true
-				} else {
-					// No requirement ID on this heading. Will be ignored.
 				}
 			}
 
-			if end {
-				// Close the current requirement.
-				newReq, err := parseReq(reqBuf.String())
-				if err != nil {
-					return nil, err
-				}
-				reqs = append(reqs, newReq)
-				inReq = false
+			// If we're currently parsing a requirement and it's appropriate close it
+			if (inReq != None) && (headingHasReqID || level < reqLevel) {
+				reqs, err = parseMarkdownFragment(inReq, reqBuf.String(), reqs)
+				inReq = None
 			}
-			if start {
-				// Start a new requirement at the current line.
-				inReq = true
+
+			// If this is the start of a new requirement initialise data
+			if headingHasReqID {
+				inReq = Heading
 				reqLevel = level
 				reqLine = lno
 				reqBuf.Reset()
 				line = title
 			}
+			if level > 0 {
+				lastHeadingLevel = level
+				lastHeadingLine = lno
+			}
+		} else if reTableHeader.MatchString(line) {
+			// It's a requirements table
+			// If we're currently parsing a requirement close it
+			if inReq != None {
+				reqs, err = parseMarkdownFragment(inReq, reqBuf.String(), reqs)
+			}
+			// Start a new requirement table
+			inReq = Table
+			reqBuf.Reset()
 		}
-		if inReq {
+
+		if inReq != None {
 			reqBuf.WriteString(line)
 			reqBuf.WriteString("\n")
 		}
-		if level > 0 {
-			lastHeadingLevel = level
-			lastHeadingLine = lno
-		}
 	}
 	if err := scan.Err(); err != nil {
 		return nil, err
 	}
 
-	if inReq {
+	if inReq != None {
 		// Close the current requirement, we're at the end.
-		newReq, err := parseReq(reqBuf.String())
+		reqs, err = parseMarkdownFragment(inReq, reqBuf.String(), reqs)
+	}
+
+	return reqs, nil
+}
+
+// parseMarkdownFragment accepts a string containing either an ATX requirement or a requirements table and calls the
+// appropriate parsing function
+func parseMarkdownFragment(reqType ReqType, txt string, reqs []*Req) ([]*Req, error) {
+
+	if reqType == Heading {
+		// An ATX requirement
+		newReq, err := parseReq(txt)
 		if err != nil {
-			return nil, err
+			return reqs, err
 		}
 		reqs = append(reqs, newReq)
+	} else {
+		var err error
```

Comment:
either move this before the if, or use := in parseReqTable.

Reply:
ahh yes, this was my ugly workaround to go creating a new reqs variable within the scope of the 'else' and then discarding it. I've used := now and a hopefully less ugly / more idiomatic way